packages/rum-core/src/domain/rumEventsCollection/action/trackActions.ts

```diff
@@ -139,6 +147,17 @@ function onClick(state: TrackActionsState, event: MouseEvent & { target: Element
     startClocks,
   })
 
+  let onEndClick: () => void
+  if (state.collectFrustrations) {
+    // If we collect frustration, we have to use a "click chain", and delay the action notification
+    // until we know that it's not part of a rage click
+    const clickReference = addClickToClickChain(state, singleClickPotentialAction)
+    onEndClick = clickReference.markAsComplete
+  } else {
+    // Else, just notify the action when on click end
+    onEndClick = singleClickPotentialAction.notifyIfComplete
+  }
+
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏to ease the reading it could be interesting to move this code after `endClick` definition